Algorithms/Search/Substring/BoyerMoore.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+
+namespace Algorithms.Search.Substring
+{
+    /// <summary>
+    /// The idea:   You compare the pattern with the text from right to lef.
+    ///             If the text symbol that is compared with the rightmost pattern symbol
+    ///             does not occur in the pattern at all, then the pattern can be shifted
+    ///             by m positions behind this text symbol.
+    /// Source: https://www.inf.hs-flensburg.de/lang/algorithmen/pattern/bmen.htm
+    /// Complexity:
+    ///     Time:   Preprocessing: O(a + m²)
+    ///             Comparison: O(n/m)
+    ///     Space:  O(m + a)
+    ///     where   m - pattern length
+    ///             n - text length
+    ///             a - alphabet length.
+    /// </summary>
+    public class BoyerMoore
```

Comment:
Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6277998)